wolfcrypt/src/asn.c

```diff
@@ -15146,6 +15194,19 @@ int DecodeToKey(DecodedCert* cert, int verify)
     int badDate = 0;
     int ret;
 
+#if defined(HAVE_RPK)
+
+    /* Raw Public Key certificate has only a SubjectPublicKeyInfo structure
+     * as its contents. So try to call GetCertKey to get public key from it.
+     * If it fails, the cert should be a X509 cert and proceed to process as
+     * x509 cert. */
+    ret = GetCertKey(cert, cert->source, &cert->srcIdx, cert->maxIdx);
+    if (ret == 0) {
+        cert.isRPK = 1;
```

Comment:
@TakayukiMatsuo should this be `cert->isRPK` ?

I was testing the RPK feature in RIOT-OS and I got the following error: `wolfssl/wolfcrypt/src/asn.c:15205:13: error: 'cert' is a pointer; did you mean to use '->'?`

Reply:
In fact, I realized that the RPK configuration does not work when `WOLFSSL_ASN_TEMPLATE` is defined:

```bash
$ make && ./tests/unit.test -test_tls13_rpk_handshake
...
processing certificate
wolfSSL Entering DoTls13Certificate
wolfSSL Entering ProcessPeerCerts
Loading peer's cert chain
        Parsing 0 bytes of cert extensions
        Put another cert into chain
Verifying Peer's cert
wolfSSL Entering GetAlgoId
wolfSSL Entering GetObjectId
wolfSSL Entering GetObjectId
Parsed Past Key
Buffer too small for ASN tag
Got Peer cert ASN PARSE_E, BUFFER E, MEMORY_E
wolfSSL Leaving ProcessPeerCerts, return -140
wolfSSL Leaving DoTls13Certificate, return -140
wolfSSL Leaving DoTls13HandShakeMsgType(), return -140
wolfSSL Leaving DoTls13HandShakeMsg, return -140
wolfSSL error occurred, error = -140
wolfSSL error occurred, error = -140
wolfSSL Entering wolfSSL_get_error
wolfSSL Leaving wolfSSL_get_error, return -140
 failed (  0.00419)

ERROR - tests/api.c line 65599 failed with:
    expected: Test failed

    result:   ret 0

wolfSSL Entering wolfSSL_Cleanup
wolfSSL Entering wolfCrypt_Cleanup

FAILURES:
   816: test_tls13_rpk_handshake

 End API Tests
```

Reply:
Hi @geonnave, thank you for heads-up. Fixed what you pointed out and addressed the case when WOLFSSL_ASN_TEMPLATE is not defined.